### DISCUSSION TITLE
'Getter' concept?

### DISCUSSION THREAD

**Discussion by gplusdotgr** [ai/nanostores#271] [General]
Is it possible to get a portion of the state based on user input (e.g. a form) without losing the original state?
vuex implements this with a 'getter' concept. 
Computed seems to be the way but it doesn't seem to accept any custom arguments?
Plus you wouldn't want to subscribe to a different store to filter the original store..

**Comment by ai** [comment]
Maybe @euaaaio can help as Vue user?

**Reply by gplusdotgr** [reply]
I finally handled it by creating two extra stores. 
1) the user input filters ( $filter ) 
2) the filtered version of my store ( $matchingOffers )
Then I used `computed` to calculate $matchingOffers based on $offer and $filter . 
Probably a little more verbose than a getter concept but does the job. Definitely a proposal for a feature :)
